lib/node_modules/@stdlib/math/base/special/ceiln/src/ceiln.c

```diff
@@ -0,0 +1,95 @@
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/math/base/special/ceiln.h"
+#include "stdlib/math/base/special/abs.h"
+#include "stdlib/constants/float64/max_base10_exponent.h"
+#include "stdlib/constants/float64/min_base10_exponent.h"
+#include "stdlib/constants/float64/max.h"
+#include <stdint.h>
+#include <math.h>
+
+/**
+* Rounds a numeric value to the nearest multiple of \\(10^n\\) toward positive infinity.
+*
+* @param x       input value
+* @param n       integer power of 10
+* @return        rounded value
+*
+* @example
+* // Round a value to 2 decimal places:
+* double y = stdlib_base_ceiln( 3.141592653589793, -2 );
+* // returns 3.15
+*
+* @example
+* // If n = 0, `ceiln` behaves like `ceil`:
+* double y = stdlib_base_ceiln( 3.141592653589793, 0 );
+* // returns 4.0
+*
+* @example
+* // Round a value to the nearest thousand:
+* double y = stdlib_base_ceiln( 12368.0, 3 );
+* // returns 13000.0
+*/
+double stdlib_base_ceiln( const double x, const int32_t n ) {
+    double s;
+    double y;
+
+    if ( isnan( x ) ){
+        return NAN;
+    }
+
+    if (
+        // Handle infinities...
+        isinf( x ) ||
+
+        // Handle +-0...
+        x == 0.0 ||
+
+        // If `n` exceeds the maximum number of feasible decimal places (such as with subnormal numbers), nothing to round...
+        n < STDLIB_CONSTANT_FLOAT64_MIN_BASE10_EXPONENT ||
+
+        // If `|x|` is large enough, no decimals to round...
+        ( stdlib_base_abs( x ) > STDLIB_CONSTANT_FLOAT64_MAX && n <= 0 ) ){
+        return x;
+    }
+
+    // The maximum absolute double is ~1.8e308. Accordingly, any possible positive finite `x` rounded to the nearest >=10^309 is infinity and any negative finite `x` is zero.
+    if ( n > STDLIB_CONSTANT_FLOAT64_MAX_BASE10_EXPONENT ) {
+        if ( x <= 0.0 ) {
+            return -0.0; // preserve the sign (same behavior as ceil)
+        }
+        return INFINITY;
+    }
+
+    // If we overflow, return `x`, as the number of digits to the right of the decimal is too small (i.e., `x` is too large / lacks sufficient fractional precision) for there to be any effect when rounding...
+    if ( n < STDLIB_CONSTANT_FLOAT64_MAX_BASE10_EXPONENT ) {
+        s = pow( 10.0, -( n + STDLIB_CONSTANT_FLOAT64_MAX_BASE10_EXPONENT ) ); // TODO: replace use of `pow` once have stdlib equivalent
+        y = ( x * HUGE ) * s; // order of operation matters!
+        if ( isinf( y ) ) {
+            return x;
+        }
+        return ( ceil( y ) / HUGE ) / s;
```

Comment:
What do we use for `HUGE` in JavaScript?

Reply:
Rather than use C `HUGE`, we should use a static variable in the parent scope, similar to the JavaScript implementation.

Reply:
@steff456 I believe `HUGE` still needs to be defined in the parent scope.